File: front/src/dto/cetificate/certificateRequestDto.ts
import { CertificateType } from './CertificateType';

export interface CertificateRequestDto {
  commonName: string; // CN
  organization: string; // O
  organizationalUnit: string; // OU
  country: string; // C
  email: string; // E
  validFrom: string;
  validTo: string;
  parentId?: string | null;
  issuerId?: string | null;
  type: CertificateType; // ROOT / INTERMEDIATE / END_ENTITY
  isCa: boolean; // da li je CA sertifikat
}

File: back/src/main/java/rs/map/pki/dto/CertificateRequestDto.java
package rs.map.pki.dto;

import lombok.Data;
import rs.map.pki.model.CertificateType;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.UUID;

@Data
public class CertificateRequestDto {
    private String commonName;  // CN
    private String organization;  // O
    private String organizationalUnit;  // OU
    private String country;  // C
    private String email;  // E
    private LocalDate validFrom;
    private LocalDate validTo;
    private UUID parentId;
    private CertificateType type; // ROOT / INTERMEDIATE / END_ENTITY
    private UUID issuerId; // ko potpisuje (CA)
    private boolean isCa;
}
